pkg/report/model/junit.go

```diff
@@ -0,0 +1,107 @@
+package model
+
+import (
+	"encoding/xml"
+	"fmt"
+
+	"github.com/Checkmarx/kics/internal/constants"
+	"github.com/Checkmarx/kics/pkg/model"
+)
+
+type junitTestSuites struct {
+	XMLName    xml.Name         `xml:"testsuites"`
+	Name       string           `xml:"name,attr"`
+	Time       string           `xml:"time,attr"`
+	Failures   string           `xml:"failures,attr"`
+	TestSuites []junitTestSuite `xml:"testsuite"`
+}
+
+type junitTestSuite struct {
+	XMLName   xml.Name        `xml:"testsuite"`
+	Name      string          `xml:"name,attr"`
+	Failures  string          `xml:"failures,attr"`
+	Tests     string          `xml:"tests,attr"`
+	TestCases []junitTestCase `xml:"testcase"`
+	failCount int
+}
+
+type junitTestCase struct {
+	XMLName  xml.Name       `xml:"testcase"`
+	Name     string         `xml:"name,attr"`
+	Failures []junitFailure `xml:"failure"`
+}
+
+type junitFailure struct {
+	XMLName xml.Name `xml:"failure"`
+	Type    string   `xml:"type,attr"`    // Query name
+	Message string   `xml:"message,attr"` // File name + line number
+}
+
+// JUnitReport is a JUnit report representation
+type JUnitReport interface {
+	GenerateTestEntry(query *model.QueryResult)
+	FinishReport()
+}
+
+// NewJUnitReport creates a new JUnit report instance
+func NewJUnitReport(time string) JUnitReport {
+	return &junitTestSuites{
+		Name:       fmt.Sprintf("KICS v%s", constants.Version),
+		Time:       time,
+		Failures:   "",
+		TestSuites: make([]junitTestSuite, 0),
+	}
+}
+
+// GenerateTestEntry generates a new test entry for failed tests on KICS scan
+func (jUnit *junitTestSuites) GenerateTestEntry(query *model.QueryResult) {
+	queryDescription := query.Description
+	if query.CISDescriptionText != "" {
+		queryDescription = query.CISDescriptionText
+	}
+	failedTestCase := junitTestCase{
+		Name:     fmt.Sprintf("[%s]: %s", query.QueryName, queryDescription),
+		Failures: make([]junitFailure, len(query.Files)),
+	}
+
+	for idx := range query.Files {
+		failedTest := junitFailure{
+			Type: query.QueryName,
+			Message: fmt.Sprintf(
+				"A problem was found on '%s' file in line %d, %s, but %s.",
```

Comment:
I suggest something like: `"A problem was found on '%s' file, in line %d. The actual value is %s. However, the expected value is %s.,` or `[file: %s, line: %d] The actual value is %s. However, the expected value is %s`.

Reply:
KICS already gives a descriptive message of what is  expected and what is the actual value, so with some "prefix" on message can make it strange, since it will look something like: "The actual value is 'The attribute <some attribute here> is <wrong value>'. However, the expected value is 'The attribute <some attribute here> is expected to be <correct value>'...
This is why I think the pattern `<expected>, but <actual>` works fine here.